### PR TITLE
Orestis us 20

### PR DESCRIPTION
**Pull request by OrestisBollano**

### CODE REVIEW

**Review by wlany — APPROVED** [review]
I got Red Kidney Bean Jambalaya!! fancy ;)

File: src/main/java/ch/uzh/ifi/hase/soprafs23/SpooncularAPI/APIController.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        Group group = groupService.getGroupById(groupId);
+
+        // 401 - not authorized
+        Long tokenId = userService.getUseridByToken(request.getHeader("X-Token"));
+        if (tokenId == 0L) {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "You are not authorized.");
+        }
+
+        Long hostId = group.getHostId();
+        User host = userService.getUserById(hostId); // Assuming there's a method to get user by Id in userService
```
Comment (akkuebler) [inline src/main/java/ch/uzh/ifi/hase/soprafs23/SpooncularAPI/APIController.java:50]: would delete the comment here, bc there is one?

File: src/test/java/ch/uzh/ifi/hase/soprafs23/controller/APIControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    @Test
+    public void getRandomRecipe_success() throws Exception {
+        mockMvc.perform(get("/groups/{groupId}/result", 1L)
+                        .header("X-Token", "valid-token")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+    }
+    @Test
+    public void getHostRecipe_returnsConflictStatusCode_whenNoRecipesFound() {
```
Comment (akkuebler) [inline src/test/java/ch/uzh/ifi/hase/soprafs23/controller/APIControllerTest.java:93]: could you pls add the error codes either within the test or next to it as comments? Helps reading the code :)

**Review by akkuebler — COMMENTED** [review]
code-wise it looks good to me

File: src/main/java/ch/uzh/ifi/hase/soprafs23/SpooncularAPI/APIController.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        Recipe detailedRecipe = apiService.getHostRecipe(host);
+
+        return String.format("ID: %d%nTitle: %s%nReady in minutes: %d%nPrice per serving: %.2f",
+                detailedRecipe.getId(),
+                detailedRecipe.getTitle(),
+                detailedRecipe.getReadyInMinutes(),
+                detailedRecipe.getPricePerServing());
+    }
```
Comment (chiaralentsch) [inline src/main/java/ch/uzh/ifi/hase/soprafs23/SpooncularAPI/APIController.java:58]: Delete string. Frontend needs the title, minutes and price per serving as parameters not as text :)

**Review by chiaralentsch — CHANGES_REQUESTED** [review]

**Pull request merged**